qemu/tests/create_external_snapshot_via_driver.py

```diff
@@ -0,0 +1,119 @@
+    3). Create external snapshot using ssh/https driver
+    4). Boot up vm using the external snapshot
+    5). Calculate the file's md5 value again and then compare
+
+    :param test: Kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def clean_key():
+
+        """
+        clear the ssh key
+        :return: None
+        """
+        process.system("rm -rf {}/id*".format(
+            params["ssh_key_dir"]), shell=True)
+
+    @error_context.context_aware
+    def create_ssh_test():
+
+        """
+        Prepare ssh environment
+        :return: None
+        """
+        error_context.context("STEP 1: Prepare ssh login environment.")
+        clean_key()
+        copy_pub_key_cmd = params.get("copy_pub_key_cmd")
+        logging.info("Create ssh key & copy it using ssh-copy-id")
+        process.system(params["create_ssh_key_cmd"], shell=True)
+        logging.info("Copy the ssh public key to remote host")
+        process.system(copy_pub_key_cmd, shell=True)
+
+    @error_context.context_aware
+    def create_https_test():
+
+        """
+        Prepare https environment
+        :return: None
+        """
+        error_context.context("STEP 1: Prepare https login environment.")
+        process.system("yum install -y mod_ssl")
+        logging.info("Start httpd service")
+        process.system(params.get("http_conf_cmd"), shell=True)
+        process.system(params.get("httpd_start_cmd"), shell=True)
+        out = process.system_output(params.get("httpd_status_cmd"), shell=True)
+        if not re.search(params.get("httpd_status_re"), out):
+            err = "Http server fails to start, because of {}".format(out)
+            exceptions.TestError(err)
+
+    subcommand = params.get("subcommand")
+    eval("%s_test()" % subcommand)
+
+    user_profile = ""
+    user_profile_backup = ""
+    if params.get("external_snapshot_protocol") == "ssh":
+        user_profile = params.get("user_profile")
+        user_profile_backup = params.get("user_profile_backup")
+        shutil.copy(user_profile, user_profile_backup)
+
+    base_image = params.get("images", "image1").split()[0]
+    params.update(
+        {"image_name_%s" % base_image: params["image_name"],
+         "image_format_%s" % base_image: params["image_format"]})
+    t_file = params["guest_file_name_%s" % base_image]
+    snapshot_test = qemu_disk_img.QemuImgTest(
+        test, params, env, base_image)
+    logging.info("STEP 2: Save file md5sum before create snapshot.")
+    snapshot_test.start_vm(params)
+    md5_before_snapshot = snapshot_test.save_file(t_file)
+    if not md5_before_snapshot:
+        raise exceptions.TestError("Fail to save tmp file.")
+    snapshot_test.destroy_vm()
+
+    logging.info("STEP 3: Start to create snapshot")
+    snapshot_test_sn = qemu_disk_img.QemuImgTest(test, params, env, "sn1")
```

Comment:
/bin/qemu-img create -f qcow2 -b /usr/share/avocado/data/avocado-vt/images/win7-32-sp1-virtio.qcow2 -F qcow2 /usr/share/avocado/data/avocado-vt/images/sn1.qcow2 30G

the cmd is not the same as expected